src/components/TodoList.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import { FC, useState } from 'react';
+import { MakeChange } from '../types/MakeChange';
+import { Todo } from '../types/Todo';
+import { TodoInfo } from './TodoInfo';
+
+type Props = {
+  todos: Todo[],
+  setTodos: MakeChange,
+  isCompletedTodo: boolean,
+};
+
+export const TodoList: FC<Props> = ({
+  todos,
+  setTodos,
+  isCompletedTodo,
+}) => {
+  const [changeTitle, setChangeTitle] = useState<number | null>(null);
+
+  const hendlerTogglerAll = () => {
+    let allCompletedTodos = todos
+      .filter(todo => !todo.completed);
+
+    if (!allCompletedTodos.length) {
+      allCompletedTodos = [...todos];
+    }
+
+    setTodos.toggle(allCompletedTodos.map(todo => (
+      { ...todo, completed: !todo.completed }
+    )));
+  };
+
+  return (
+    <section className="main">
+      <input
+        type="checkbox"
+        id="toggle-all"
+        className="toggle-all"
+        data-cy="toggleAll"
+        checked={isCompletedTodo}
+        onClick={hendlerTogglerAll}
```

Comment:
Use 'onChange' event instead of 'onClick' for better accessibility of the input checkbox.